dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -512,7 +512,13 @@ private boolean isMemberAccessibleFrom(
       }
 
       if (!isClassAccessibleFrom(targetClass, sourceClassName)) {
-        return Optional.of(new InaccessibleClassProblem(sourceClassFile, targetClassFile, symbol));
+        if (classDumper.isUnusedClassSymbolReference(sourceClassName, symbol)) {
```

Comment:
What about:

```java
if (!isClassAccessibleFrom(targetClass, sourceClassName)
    && !classDumper.isUnusedClassSymbolReference(sourceClassName, symbol)) {
  return Optional.of(new InaccessibleClassProblem(sourceClassFile, targetClassFile, symbol));
}
return Optional.empty();
```

Reply:
I tried that and I was confused. Maybe this is more clear:

```
if (isClassAccessibleFrom(targetClass, sourceClassName)
    || classDumper.isUnusedClassSymbolReference(sourceClassName, symbol)) {
  return Optional.empty();
} else {
  return Optional.of(new InaccessibleClassProblem(sourceClassFile, targetClassFile, symbol));
}
```

Reply:
Maybe, but it does seem like an inversion of the test.

If the class is inaccessible and it's not unused... What if you inverted `isUnusedClassSymbolReference`?

```java
if (!isClassAccessibleFrom(targetClass, sourceClassname)
    && classDumpter.isClassSymbolReferenceUsed(sourceClassName, symbol)) {
  return Optional.of(…);
}
return Optional.empty();
```

Reply:
Added isClassSymbolReferenceUsed to ClassDumper.